velocity-custom-parser-example/pom.xml

```diff
@@ -0,0 +1,267 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.velocity</groupId>
+        <artifactId>velocity-engine-parent</artifactId>
+        <version>2.2-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>velocity-custom-parser-example</artifactId>
+    <name>Apache Velocity Custom Parser - Example</name>
+    <description>Custom Parser Example for Apache Velocity</description>
+
+    <packaging>jar</packaging>
+
+    <!--
+      This plugin demonstrates how to build a custom Velocity parser.
+      The proposed custom parser replaces '#' with '@' and '@' with '%',
+      so that it's suitable to use with Markdown template files, for instance.
+
+      The generated parser class is org.apache.velocity.runtime.parser.${parser.name}Parser,
```

Comment:
I have a problem with that. I don't people to create there parsers in our namespace. I'd expect the name/prefix to be decoupled form the package name: `parser.name`, `parser.package`.

Reply:
I don't like it either, but it's a technical constraint. Javacc is generating a bunch of side classes, some of which must be discarded in favor of our customized versions, which must be in the same package, hence `org.apache.velocity.runtime.parser`, which will remain the package of all javacc-generated parsers.

Reply:
Looks like it's doable after all with a little more work. Done.